File: Backend/LinkCare/src/main/java/com/ssafy/linkcare/group/dto/WeeklyGroupGoalResponse.java
package com.ssafy.linkcare.group.dto;

import com.ssafy.linkcare.group.entity.WeeklyGroupGoals;
import com.ssafy.linkcare.group.enums.MetricType;

import java.time.LocalDate;

/**
 * 주간 그룹 목표 응답 DTO
 */
public record WeeklyGroupGoalResponse(
        Integer weeklyGroupGoalsSeq,
        Long groupSeq,
        LocalDate weekStart,
        Long goalSteps,
        Float goalKcal,
        Integer goalDuration,
        Float goalDistance,
        Double predictedGrowthRateSteps,
        Double predictedGrowthRateKcal,
        Double predictedGrowthRateDuration,
        Double predictedGrowthRateDistance,
        MetricType selectedMetricType
) {
    public static WeeklyGroupGoalResponse from(WeeklyGroupGoals goal) {
        return new WeeklyGroupGoalResponse(
                goal.getWeeklyGroupGoalsSeq(),
                goal.getGroup().getGroupSeq(),
                goal.getWeekStart(),
                goal.getGoalSteps(),
                goal.getGoalKcal(),
                goal.getGoalDuration(),
                goal.getGoalDistance(),
                goal.getPredictedGrowthRateSteps(),
                goal.getPredictedGrowthRateKcal(),
                goal.getPredictedGrowthRateDuration(),
                goal.getPredictedGrowthRateDistance(),
                goal.getSelectedMetricType()
        );
    }
}

File: Frontend/app/src/main/java/com/a307/linkcare/feature/healthgroup/data/model/response/WeeklyGroupGoalResponse.kt
package com.a307.linkcare.feature.healthgroup.data.model.response

data class WeeklyGroupGoalResponse(
    val weeklyGroupGoalsSeq: Long,
    val groupSeq: Long,
    val weekStart: String,  // "2024-01-08"
    val goalSteps: Long,
    val goalKcal: Float,
    val goalDuration: Int,  // minutes
    val goalDistance: Float,  // km
    val predictedGrowthRateSteps: Double,
    val predictedGrowthRateKcal: Double,
    val predictedGrowthRateDuration: Double,
    val predictedGrowthRateDistance: Double,
    val selectedMetricType: String? = null  // "STEPS", "KCAL", "DURATION", "DISTANCE"
)
